### PR TITLE
Change Authenticator name to GCP

### PR DESCRIPTION
**Pull request by nessiLahav**
### What does this PR do?
Change the name of GCE back to GCP, in order to future support app engine and cloud functions use cases with no need to create a new api name

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 ### Added
-- Hosts can authenticate from Google Compute Engines (GCE) using a GCE instance
-  identity token. See [design](design/authenticators/authn_gce/authn_gce_solution_design.md)
+- Hosts can authenticate from Google Compute Engines (GCP) using a GCP instance
```
Comment (Tovli) [inline CHANGELOG.md:9]: Google compute engine is still GCE

Comment (nessiLahav) [reply CHANGELOG.md:9]: Nice catch 👍

File: Jenkinsfile [hunk 2]
```diff
@@ -97,13 +97,13 @@ pipeline {
             }
           }
         }
-        // We have 2 stages for GCE Authenticator tests. The first one runs inside
-        // a GCE instance and retrieves all the tokens that will be used in the tests.
+        // We have 2 stages for GCP Authenticator tests. The first one runs inside
+        // a GCP instance and retrieves all the tokens that will be used in the tests.
```
Comment (Tovli) [inline Jenkinsfile:101]: GCE instance

File: Jenkinsfile [hunk 3]
```diff
@@ -97,13 +97,13 @@ pipeline {
...
         // It then stashes the tokens, which are unstashed in the stage that runs the
-        // GCE Authenticator tests using the tokens.
-        // This way we can have a light-weight GCE instance that has no need for conjurops
+        // GCP Authenticator tests using the tokens.
+        // This way we can have a light-weight GCP instance that has no need for conjurops
```
Comment (Tovli) [inline Jenkinsfile:104]: GCE instance

File: Jenkinsfile [hunk 4]
```diff
@@ -97,13 +97,13 @@ pipeline {
...
         // or git identities and is not open for SSH
-        stage('GCE Authenticator preparation - Allocate GCE Instance') {
+        stage('GCP Authenticator preparation - Allocate GCP Instance') {
```
Comment (Tovli) [inline Jenkinsfile:106]: the second GCP should be GCE

**Review by eranha — DISMISSED** [review]

**Review by eranha — APPROVED** [review]

**Pull request merged**